commands/command_filter.go

```diff
@@ -138,29 +160,32 @@ func filterCommand(cmd *cobra.Command, args []string) {
 
 Scan:
 	for s.Scan() {
-		req := s.Request()
-
-		// TODO:
-		// clean/smudge should also take a Writer instead of returning []byte
-		var outputData []byte
 		var err error
+		var w io.Writer
 
-		switch req.Header["command"] {
+		switch req := s.Request(); req.Header["command"] {
 		case "clean":
-			outputData, err = clean(req.Payload, req.Header["pathname"])
+			w = git.NewPacketWriter(os.Stdout, cleanFilterBufferCapacity)
+			err = clean(w, req.Payload, req.Header["pathname"])
 		case "smudge":
-			outputData, err = smudge(req.Payload, req.Header["pathname"])
+			w = git.NewPacketWriter(os.Stdout, smudgeFilterBufferCapacity)
+			err = smudge(w, req.Payload, req.Header["pathname"])
 		default:
 			fmt.Errorf("Unknown command %s", cmd)
 			break Scan
 		}
 
+		if err == nil {
+			_, err = w.Write(nil)
+		}
+
+		var status string
 		if err != nil {
-			s.WriteStatus("error")
+			status = "error"
 		} else {
-			s.WriteStatus("success")
-			s.WriteResponse(outputData)
+			status = "success"
 		}
+		s.WriteStatus(status)
```

Comment:
I think this was originally incorrect but has since been fixed in https://github.com/github/git-lfs/pull/1622/commits/4c1615432a126826717d6a632486198aad8faaec and https://github.com/github/git-lfs/pull/1622/commits/10660fc0b8f3eee4fa77a7979470d9de4496ee50.